tests/components/irm_kmi/conftest.py

```diff
@@ -0,0 +1,134 @@
+"""Fixtures for the IRM KMI integration tests."""
+
+from __future__ import annotations
+
+from collections.abc import Generator
+import json
+from unittest.mock import MagicMock, patch
+
+from irm_kmi_api import IrmKmiApiError
+import pytest
+
+from homeassistant.components.irm_kmi.const import DOMAIN
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE, CONF_LOCATION
+
+from tests.common import MockConfigEntry, load_fixture
+
+
+@pytest.fixture
+def mock_config_entry() -> MockConfigEntry:
+    """Return the default mocked config entry."""
+    entry = MockConfigEntry(
+        title="Home",
+        domain=DOMAIN,
+        data={CONF_LOCATION: {ATTR_LATITUDE: 50.84, ATTR_LONGITUDE: 4.35}},
+        unique_id="50.84-4.35",
+    )
+    entry.runtime_data = MagicMock()
+    return entry
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[None]:
+    """Mock setting up a config entry."""
+    with patch("homeassistant.components.irm_kmi.async_setup_entry", return_value=True):
+        yield
+
+
+@pytest.fixture
+def mock_get_forecast_in_benelux():
+    """Mock a call to IrmKmiApiClient.get_forecasts_coord() so that it returns something valid and in the Benelux."""
+    with patch(
+        "homeassistant.components.irm_kmi.config_flow.IrmKmiApiClient.get_forecasts_coord",
+        return_value={"cityName": "Brussels"},
+    ):
+        yield
+
+
+@pytest.fixture
+def mock_get_forecast_out_benelux():
+    """Mock a call to IrmKmiApiClient.get_forecasts_coord() so that it returns something outside Benelux."""
+    with patch(
+        "homeassistant.components.irm_kmi.config_flow.IrmKmiApiClient.get_forecasts_coord",
+        return_value={"cityName": "Outside the Benelux (Brussels)"},
+    ):
+        yield
+
+
+@pytest.fixture
+def mock_get_forecast_api_error():
+    """Mock a call to IrmKmiApiClient.get_forecasts_coord() so that it raises an error."""
+    with patch(
+        "homeassistant.components.irm_kmi.config_flow.IrmKmiApiClient.get_forecasts_coord",
+        side_effet=IrmKmiApiError,
+    ):
+        return
+
+
+@pytest.fixture
+def mock_get_forecast_api_error_repair():
+    """Mock a call to IrmKmiApiClient.get_forecasts_coord() so that it raises an error."""
+    with patch(
+        "homeassistant.components.irm_kmi.repairs.IrmKmiApiClient.get_forecasts_coord",
+        side_effet=IrmKmiApiError,
+    ):
+        return
```

Comment:
No longer used? In the other case you may want to set side_effect instead of side_effet.

Reply:
Thanks for checking the unused fixtures.  Indeed, not needed anymore.  Removed in 94aa3ca35db19f6bd33c5ff601d269efba2ff33f